packages/jest-editor-support/src/SnapshotSupport.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+const babylon = require('babylon');
+const fs = require('fs');
+const path = require('path');
+const walker = require('./walker');
+
+type Node = any;
+
+type SnapshotMetadata = {
+  exists: true | false,
+  exportName: string,
+  node: Node,
+  content?: string,
+};
+
+type SnapshotSupportType = {
+  getMetadataFromTestFile: (filePath: string) => Array<SnapshotMetadata>,
+  parser?: (content: string, options: any) => Node,
+  plugins?: Array<string>,
+};
+
+const buildExportName: (
+  toMatchSnapshot: Node,
+  parents: Array<Node>
+) => string = (toMatchSnapshot, parents) => {
+  const state = {
+    found: [],
+  };
+
+  walker(
+    parents[parents.length - 1],
+    {
+      Identifier(node, state, parents) {
+        if (node.name === 'toMatchSnapshot') {
+          state.found.push({node, parents});
+        }
+      },
+    },
+    state
+  );
```

Comment:
Oh, I see now. You are using the walker to descend into `toMatchSnapshot` first and then you are going back up to find the name, right? Is there a way we could track the parent nodes when we descend into the AST so that we don't have to go down and then back up the AST (for perf reasons)?